### PR TITLE
Add encap test cases

### PR DESCRIPTION
**Pull request by xw-g**

### CODE REVIEW

File: feature/gribi/basic_encap/README.md [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+
+*   All egress packets (100%) are IPinIP (4in4) encapped.
+*   Packets are encapped to the tunnel IPs in the specified ratio. Specifically,
+    25% of the egress packets should have the destination address 203.0.113.1,
+    and 75% of the egress packets should have the destination address
+    203.10.113.2.
+*   The encapped/tunneled packets should be distributed hierarchically per the
+    weight.
+*   The DSCP value is copied from the inner header to the outer header.
+*   The TTL value is copied from the inner header to the outer header.
```
Comment (singh-prem) [inline feature/gribi/basic_encap/README.md:311]: Can you confirm if this is an expected requirement. As per my understanding the inner TTL value should not be copied to outer header.

File: feature/gribi/encap_frr/README.md [hunk 2]
```diff
@@ -0,0 +1,492 @@
...
+  {NH#11, DEFAULT VRF, weight:3,mac_address:magic_mac, interface-ref:dut-port-3-interface},
+}
+IPv4Entry {192.0.2.102/32 (DEFAUlT VRF)} -> NHG#3 (DEFAULT VRF) -> {
+  {NH#100, DEFAULT VRF, weight:2,mac_address:magic_mac, interface-ref:dut-port-4-interface},
+}
+
+NHG#8 (Default VRF) {
+  {NH#1000, DEFAULT VRF}
+}
+NH#1000 -> {
```
Comment (arvbaska1) [inline feature/gribi/encap_frr/README.md:295]: Should this be a DecapEncap action for this NH ?
If NH#1 & NH#2 are down, then Encap traffic will be adding another Encap on existing encap header ?

**Review by nachikethas — APPROVED** [review]

File: feature/gribi/basic_encap/README.md [hunk 3]
```diff
@@ -0,0 +1,389 @@
+# TE-16.1: basic encapsulation tests
+
+## Summary
+
+Test basic encapsulation behaviors.
+
+## Topology
+
+ATE port-1 <------> port-1 DUT
```
Comment (manan-patel) [inline feature/gribi/basic_encap/README.md:9]: Keep same order 
DUT <-> ATE

File: feature/gribi/basic_encap/README.md [hunk 4]
```diff
@@ -0,0 +1,389 @@
+# TE-16.1: basic encapsulation tests
```
Comment (singh-prem) [inline feature/gribi/basic_encap/README.md:1]: can we move the README.md file for TE-16.1 to a folder like feature/gribi/otg_tests/basic_encap_test ? or any suitable name. I have raised a PR with implementation of this test using PR https://github.com/openconfig/featureprofiles/pull/2456
But rundata test will fail in the absence of README.md file.

Comment (xw-g) [reply feature/gribi/basic_encap/README.md:1]: feel free to move this folder along as part of your PR. I"m totally ok with that.

**Comment by xw-g** [comment]
Received internal approval from b/304783132. Further comment/questions can be raised from separate PR or bugs.

File: feature/gribi/basic_encap/README.md (hunk 1 above)
Comment (xw-g) [reply feature/gribi/basic_encap/README.md:311]: Yes. It's expected. Feel free to open a internal bug if you have further concerns. Thanks.

File: feature/gribi/encap_frr/README.md (hunk 2 above)
Comment (xw-g) [reply feature/gribi/encap_frr/README.md:295]: https://github.com/openconfig/featureprofiles/pull/2514

Comment (arvbaska1) [reply feature/gribi/encap_frr/README.md:295]: Thanks

**Pull request merged**